### PR TITLE
[1473] copy about course content

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
Copying course content

### Changes proposed in this pull request
Add "Copy from" select to enable users to copy content from another course within their org

### Guidance to review
Example - `/organisations/2AT/courses/3CXD/about`

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -1,43 +1,29 @@
...
+                             .includes(provider: [:sites])
+                             .includes(:accrediting_provider)
+                             .where(provider_code: @provider_code)
+                             .find(params[:copy_from])
+                             .first
+
+      course.about_course = @source_course.about_course
+      course.interview_process = @source_course.interview_process
+      course.how_school_placements_work = @source_course.how_school_placements_work
+    end
```
Comment (misaka) [inline app/controllers/courses_controller.rb]: :100:

File: app/views/courses/about.html.erb [hunk 2]
```diff
@@ -83,4 +104,49 @@
...
+  <div class="govuk-grid-column-one-third">
+    <aside class="related">
+      <h3 class="govuk-heading-m">Copy content from another course</h3>
+      <p class="govuk-body">Fill in this page with content from another course.</p>
+      <p class="govuk-body">This will replace any existing content.</p>
+      <%= form_for course, url: about_provider_course_path(course.provider.provider_code, course.course_code), method: :get do |f| %>
+        <div class="govuk-form-group">
+          <label class="govuk-label govuk-label--s" for="copy-from">Copy from</label>
+          <select id="copy-from" name="copy_from" class="govuk-select">
+            <option value="" selected="selected">Pick a course</option>
```
Comment (defong) [inline app/views/courses/about.html.erb]: will this get displayed when there is no courses to copy from?

**Comment by misaka** [comment]
Looking good!

**Comment by benilovj** [comment]
When there's no courses to copy from, the C# version shows the following copy:

image

File: app/views/courses/about.html.erb (hunk 2 above)
Comment (dankmitchell) [reply app/views/courses/about.html.erb]: Done

File: spec/requests/courses/about_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    end
+
+    context 'with copy_form parameter' do
+      it 'renders the course about with data from chosen' do
+        get(about_provider_course_path(provider_code: provider.provider_code, code: course.course_code, params: { copy_from: course_2.course_code }))
+
+        expect(response.body).to include(course.name)
+        expect(response.body).to include(course_2.about_course)
+        expect(response.body).to include(course_2.interview_process)
+        expect(response.body).to include(course_2.how_school_placements_work)
```
Comment (misaka) [inline spec/requests/courses/about_spec.rb]: Could we give this the `SitePrism` treatment? Or is that what @heidar is doing?

Comment (dankmitchell) [reply spec/requests/courses/about_spec.rb]: Spent 2hours on this and couldn't get it working so we can re-visit

File: spec/requests/courses/about_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      )
+    end
+
+    it 'renders the course about' do
+      get(about_provider_course_path(provider_code: provider.provider_code,
+                                     code: course.course_code))
+
+      expect(response.body).to include(
+        "#{course.name} (#{course.course_code})"
+      )
```
Comment (misaka) [inline spec/requests/courses/about_spec.rb]: Can we do this with SitePrism?

Comment (misaka) [reply spec/requests/courses/about_spec.rb]: Sorry, didn't see your response elsewhere. Ya, we'll have to re-visit this then.

**Comment by fofr** [comment]
Found a couple of problems with this on dev:

After saving, changes shouldn't be published:
Screen Shot 2019-05-17 at 09 35 29

After copying and hitting the Save button I shouldn't get this warning:
Screen Shot 2019-05-17 at 09 34 23

Can we make the background of the select white?
Screen Shot 2019-05-17 at 09 33 22

**Comment by fofr** [comment]
This line is throwing:
https://github.com/DFE-Digital/manage-courses-frontend/blob/master/app/controllers/courses_controller.rb#L21

`undefined method `reject' for nil:NilClass`
on `/organisations/2BV/courses/38ZQ/about`

**Comment by dankmitchell** [comment]
[Add sidebar with Copy functionality to "About this course" edit page](https://trello.com/c/gJQI85NG/1473-add-sidebar-with-copy-functionality-to-about-this-course-edit-page)

**Pull request merged**